pkg/jobs/hive/hive.go

```diff
@@ -329,8 +333,268 @@ func UpgradeCluster(client clientv1.Client, clusterName string, curator *cluster
 	return nil
 }
 
-func MonitorUpgradeStatus(client clientv1.Client, clusterName string, curator *clustercuratorv1.ClusterCurator) error {
+func EUSUpgradeCluster(client clientv1.Client, clusterName string, curator *clustercuratorv1.ClusterCurator, isInterVersion bool) error {
+	updateVersion := curator.Spec.Upgrade.IntermediateUpdate
+	if !isInterVersion {
+		updateVersion = curator.Spec.Upgrade.DesiredUpdate
+	}
+
+	if isInterVersion {
+		klog.V(0).Info("* Initiate EUS to EUS Intermediate Upgrade to " + updateVersion)
+	} else {
+		klog.V(0).Info("* Initiate EUS to EUS Final Upgrade to " + updateVersion)
+	}
+
+	if err := validateEUSUpgradeVersion(client, clusterName, curator, isInterVersion); err != nil {
+		return err
+	}
+
+	klog.V(2).Info("Check if managedclusterview exists " + clusterName)
+
+	// For OCP/Kubenetes versions that removed APIs, need to acknowledge this
+	// before upgrading or else upgrade will be blocked.
+	ocpConfigMCV := &managedclusterviewv1beta1.ManagedClusterView{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      clusterName + "admack",
+			Namespace: clusterName,
+			Labels: map[string]string{
+				MCVUpgradeLabel: clusterName,
+			},
+		},
+		Spec: managedclusterviewv1beta1.ViewSpec{
+			Scope: managedclusterviewv1beta1.ViewScope{
+				Kind:      "ConfigMap",
+				Name:      "admin-acks",
+				Namespace: "openshift-config",
+				Version:   "v1",
+			},
+		},
+	}
+
+	ocpConfigView := managedclusterviewv1beta1.ManagedClusterView{}
+	if err := client.Get(context.TODO(), types.NamespacedName{
+		Name:      clusterName + "admack",
+		Namespace: clusterName,
+	}, &ocpConfigView); err != nil {
+		// check if mcv exists before creating
+
+		klog.V(2).Info("Create managedclusterview " + clusterName + "admack")
+		if err := client.Create(context.TODO(), ocpConfigMCV); err != nil {
+			return err
+		}
+	}
+
+	ocpConfigGetErr := errors.New("Failed to get remote admin-ack configmap")
+	resultOCPConfigMCV := managedclusterviewv1beta1.ManagedClusterView{}
+	if err := waitForMCV(client, clusterName+"admack", clusterName, &resultOCPConfigMCV, ocpConfigGetErr); err != nil {
+		return err
+	}
+
+	resultConfigMap := resultOCPConfigMCV.Status.Result
+	configMap := map[string]interface{}{}
```

Comment:
nit. we should probably import the core api and use the configmap type here. If it's a problem then just leave it.